crates/fj-proc/src/attributed_arguments.rs

```diff
@@ -0,0 +1,198 @@
+use proc_macro::TokenStream;
+use quote::quote;
+use syn::{
+    bracketed, parenthesized, parse::Parse, parse_macro_input, parse_quote,
+};
+
+pub fn attributed_arguments(_: TokenStream, input: TokenStream) -> TokenStream {
+    let item = parse_macro_input!(input as syn::ItemFn);
+    let inputs = item.clone().sig.inputs;
+
+    let args: Vec<Argument> =
+        inputs.iter().map(|inp| parse_quote!(#inp)).collect();
+
+    let mut defaults = Vec::new();
+    let mut mins = Vec::new();
+    let mut maxs = Vec::new();
+    let mut names = Vec::new();
+    let mut types = Vec::new();
+    for arg in args {
+        let mut default = None;
+        let mut min = None;
+        let mut max = None;
+        names.push(arg.ident);
+        types.push(arg.ty);
+        for value in arg.attr.values.clone() {
+            if let Some(ident) = value.ident.clone() {
+                match ident.to_string().as_str() {
+                    "default" => default = Some(value.val.clone()),
+                    "min" => min = Some(value.val.clone()),
+                    "max" => max = Some(value.val.clone()),
+                    _ => {}
+                }
+            } else {
+                default = Some(value.val.clone());
```

Comment:
Maybe it would be better not to have this fallback, and always require an explicit `default = ` to specify default values. Would be more verbose in the simple case, but also easier to read and understand.

What do you think?

Reply:
I think I'll simplify the macro to not allow for this implicit fallback